core/src/codegen/variant_data.rs

```diff
@@ -58,11 +58,17 @@ impl<'a> FieldsGen<'a> {
 
         quote!(
             for __item in __items {
-                if let ::syn::NestedMeta::Meta(ref __inner) = *__item {
-                    let __name = ::darling::util::path_to_string(__inner.path());
-                    match __name.as_str() {
-                        #(#arms)*
-                        __other => { #handle_unknown }
+                match *__item {
+                    ::syn::NestedMeta::Meta(ref __inner) => {
+                        let __name = ::darling::util::path_to_string(__inner.path());
+                        match __name.as_str() {
+                            #(#arms)*
+                            __other => { #handle_unknown }
+                        }
+                    }
+                    ::syn::NestedMeta::Lit(ref __inner) => {
+                        __errors.push(::darling::Error::unsupported_format("literal")
+                            .with_span(__inner));
```

Comment:
Should this honor `allow_unknown_fields` and avoid an error? I'm a bit worried that this change might break a use-case where `darling` does a partial parse, and thinking that the existing flag for this would cover that scenario. 

If there are reasons we shouldn't do that, we _could_ add a new `allow_literals`, but I'd want to see more evidence of a need for another flag before we expand the API surface.

Reply:
> I'm a bit worried that this change might break a use-case where `darling` does a partial parse

Would a partial parse result in a nested meta being interpreted as a literal rather than a meta? A literal would always come after a path so it's less clear to me how we'd get a literal without the path in a partial parse.

On the one hand respecting `allow_unknown_fields` here could make sense if in the future darling is going to support literals are params.

On the other hand it doesn't really make sense because today if you pass a literal to the macro you're using the macro incorrectly.

Happy to do what you think is best here.

Reply:
I think leaving at as-shown in the PR is reasonable.